src/conn/conn_api.c

```diff
@@ -1149,6 +1150,13 @@ __conn_reconfigure(WT_CONNECTION *wt_conn, const char *config)
 	__wt_free(session, conn->cfg);
 	conn->cfg = p;
 
+	/*
+	 * Fourth, run any diagnostic commands that were included. Do that last,
+	 * reconfiguration might include other configuration changes to support
+	 * a command.
+	 */
+	WT_ERR(__conn_verbose_cmd(session, config));
```

Comment:
Would it make sense to move this into {{__wt_verbose_config}} and short-circuit there if verbose is the only option specified. I guess that requires doing some parsing of the config string, but potentially restarting all the utility threads each time an application tries to output a verbose message makes me nervous. Phrased another way: Until this point I've been thinking of reconfigure as a heavy weight operation that is likely to trigger significant changes to the running environment. This functionality is more like "dump some info" and I'd prefer if they weren't entangled.

I feel like I'm selling myself on adding a new API instead of using ``WT_CONNECTION::reconfigure``, but that has back-porting consequences and means more plumbing will need to be added to MongoDB to expose the functionality.